common/src/sql/dbinit.sql

```diff
@@ -128,6 +128,39 @@ CREATE INDEX ON omicron.public.service (
     sled_id
 );
 
+-- x509 certificates which may be used by services
+CREATE TABLE omicron.public.certificate (
+    -- Identity metadata (asset)
+    id UUID PRIMARY KEY,
+    name STRING(63) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ,
+
+    -- The service type which should use this certificate
+    service omicron.public.service_kind NOT NULL,
+
+    -- cert.pem file as a binary blob
+    cert BYTES NOT NULL,
+
+    -- key.pem file as a binary blob
+    key BYTES NOT NULL
+);
+
+-- Add an index which lets us look up certificates for a particular service
+-- class.
+CREATE INDEX ON omicron.public.certificate (
+    service
```

Comment:
I'm always worried about non-unique indexes -- see #1497.  Is this one okay?

Reply:
I can also add an index by ID, I think that should be fine?

I really just want the ability to have certificates for different classes of service (e.g., "here are all certificates for nexus") so I can filter by that category and pull them out later. This is a case that definitely shouldn't be unique - it's fine to have multiple certs for a service class - but would the following eliminate this problem?

```sql
CREATE INDEX on omicron.public.certificate (
  id,
  service
) WHERE time_deleted IS NULL;
```

(Updated in 24d9421fd07e7f7f391e5b457c4c9726ad8b8ca8)

Reply:
I think the order of columns in the index is wrong now.  I think you want it on (service, id).  Otherwise, you can't use this index to efficiently list the certificates for a particular service because they're sorted by id first.  (But you _do_ need the id in the index or else you can't _page_ through the list of certificates for a particular service using the index.)